### PR TITLE
Error handling while creating client using config file data

### PR DESCRIPTION
**Pull request by adsri**

### CODE REVIEW

File: common/config.go [hunk 1]
```diff
@@ -25,6 +25,16 @@ type GridConfig struct {
 	SslVerify           string `toml:"ssl-verify" env:"SSL_VERIFY"`
 	HttpRequestTimeout  uint   `toml:"http-request-timeout" env:"HTTP_REQUEST_TIMEOUT"`
 	HttpPoolConnections uint   `toml:"http-pool-connections" env:"HTTP_POOL_CONNECTIONS"`
+	_wapiPassword       string
+}
+
+func (gridConfig *GridConfig) SetWapiPassword() {
+	gridConfig._wapiPassword = gridConfig.WapiPassword
+	gridConfig.WapiPassword = "******"
+}
```
Comment (ishantt) [inline common/config.go]: Instead of doing this you can implement the String() method to implement the Stringer interface which returns all fields except the Password.

File: driver/ipam_driver.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func main() {
 		Version:  config.WapiVer,
 		Port:     config.WapiPort,
 		Username: config.WapiUsername,
-		Password: config.WapiPassword,
+		Password: config.SecuredWapiPassword(),
```
Comment (ishantt) [inline driver/ipam_driver.go]: Not required if you implement the String() method

File: ea-defs/create_ea_defs.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ func main() {
 		Version:  config.WapiVer,
 		Port:     config.WapiPort,
 		Username: config.WapiUsername,
-		Password: config.WapiPassword,
+		Password: config.SecuredWapiPassword(),
```
Comment (ishantt) [inline ea-defs/create_ea_defs.go]: ditto

File: vendor/github.com/infobloxopen/infoblox-go-client/connector.go [hunk 4]
```diff
@@ -331,6 +331,10 @@ func NewConnector(hostConfig HostConfig, transportConfig TransportConfig,
 	connector.Requestor.Init(connector.TransportConfig)
 
 	res = connector
```
Comment (ishantt) [inline vendor/github.com/infobloxopen/infoblox-go-client/connector.go:333]: Dont make changes directly in the vendored code. Checkout infoblox-go-client fork, make changes and then when they gets merged, update the dependencies using glide.

File: vendor/github.com/infobloxopen/infoblox-go-client/connector.go [hunk 5]
```diff
@@ -331,6 +331,10 @@ func NewConnector(hostConfig HostConfig, transportConfig TransportConfig,
...
-
```
Comment (ishantt) [inline vendor/github.com/infobloxopen/infoblox-go-client/connector.go:334]: Add a comment here stating why are you doing this.

File: vendor/github.com/infobloxopen/infoblox-go-client/connector.go [hunk 6]
```diff
@@ -331,6 +331,10 @@ func NewConnector(hostConfig HostConfig, transportConfig TransportConfig,
...
+	userprofile := NewUserProfile(UserProfile{})
+	_, err = connector.makeRequest(GET, userprofile, "")
+	if err != nil {
+		log.Printf("Failed while creating client...\n")
```
Comment (ishantt) [inline vendor/github.com/infobloxopen/infoblox-go-client/connector.go]: maybe reword as "Failed to connect to the Grid" . Also log what is the error.

**Review by ishantt — CHANGES_REQUESTED** [review]

File: common/config.go [hunk 7]
```diff
@@ -43,12 +49,22 @@ type PluginConfig struct {
 	IpamConfig `toml:"ipam-config"`
 }
 
+func (pc *PluginConfig) String() string {
+	return fmt.Sprintf("{ConfigFile: %v, Debug: %v, GridConfig: %v, IpamConfig: %v}",
+		pc.ConfigFile, pc.Debug, pc.GridConfig, pc.IpamConfig)
+}
+
```
Comment (ishantt) [inline common/config.go]: Any reason why the receiver is pointer to PluginConfig?

Comment (adsri) [reply common/config.go]: no reason,  It will work without pointer also. to make it consistent i will remove pointer as reciever.

**Review by ishantt — APPROVED** [review]
